do-not-commit/bind.py

```diff
@@ -0,0 +1,29 @@
+# import socket as s0
+# import socket.socket as s1
+# from socket import socket as s1
+
+import socket
+
+mys = s1(doesnt, matter)
+# todoruleid:avoid-bind-to-all-interfaces
+s1.bind(("", 1337))
+
+theirs = s1(doesnt, matter)
+
+# todoruleid:avoid-bind-to-all-interfaces
+theirs.bind(("0.0.0.0", 1337))
+
+# ok
+theirs.bind(("8.8.8.8", 1337))
+
+# won't work, shouldn't fire
+theirs.bind("", 1337)
+
+# easy
+
+#  (TODO-false positive)
+# ruleid:avoid-bind-to-all-interfaces
+easy_s = socket.socket(doesnt, matter)
```

Comment:
Running `socket.bind` to 0.0.0.0, ::, or empty string could unexpectedly
expose the server publicly as it binds to all available interfaces. Consider
instead getting correct address from an environment variable or
configuration file. ([dit.network.bind.avoid-bind-to-all-interfaces](https://semgrep.dev/r/dit.network.bind.avoid-bind-to-all-interfaces))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=dit.network.bind.avoid-bind-to-all-interfaces) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>